File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Mappers/Commons/ConservationModelMapper.ts
import BaseMapper = require('App/Mappers/BaseMapper');
import ConservationModel = require('App/Models/Commons/ConservationModel')
import ConservationStatusType = require('App/Models/Commons/ConservationStatusType');
import ConservationType = require('App/Models/Commons/ConservationType');

class ConservationModelMapper extends BaseMapper<ConservationModel>{
    constructor() {
        super();
    }

    public Map(source: any): ConservationModel {
        let toMap = <ConservationModel>{};

        if (!source) {
            return null;
        }

        toMap.EntityType = source.EntityType;
        toMap.Message = source.Message;
        toMap.SendDate = source.SendDate;
        toMap.Status = ConservationStatusType[source.Status as string];
        toMap.Type = ConservationType[source.Type as string];
        toMap.UniqueId = source.UniqueId;
        toMap.Uri = source.Uri;
        return toMap;
    }
}

export = ConservationModelMapper;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Model/Conservations/Implementation/ConservationModelMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.Conservations;
using VecompSoftware.DocSuiteWeb.Model.Entities.Conservations;

namespace VecompSoftware.DocSuiteWeb.Mapper.Model.Conservations
{
    public class ConservationModelMapper : BaseModelMapper<Conservation, ConservationModel>, IConservationModelMapper
    {
        #region [ Fields ]
        private readonly IMapperUnitOfWork _mapperUnitOfWork;
        #endregion

        #region [ Constructor ]
        public ConservationModelMapper(IMapperUnitOfWork mapperUnitOfWork)
        {
            _mapperUnitOfWork = mapperUnitOfWork;
        }
        #endregion

        #region [ Methods ]

        public override ConservationModel Map(Conservation entity, ConservationModel modelTransformed)
        {
            modelTransformed.EntityType = entity.EntityType;
            modelTransformed.Message = entity.Message;
            modelTransformed.SendDate = entity.SendDate;
            modelTransformed.Status = (DocSuiteWeb.Model.Conservations.ConservationStatus)entity.Status;
            modelTransformed.Type = (DocSuiteWeb.Model.Conservations.ConservationType)entity.Type;
            modelTransformed.Uri = entity.Uri;
            modelTransformed.UniqueId = entity.UniqueId;
            modelTransformed.RegistrationDate = entity.RegistrationDate;
            modelTransformed.RegistrationUser = entity.RegistrationUser;
            modelTransformed.LastChangedDate = entity.LastChangedDate;
            modelTransformed.LastChangedUser = entity.LastChangedUser;

            return modelTransformed;
        }
        #endregion
    }
}
